### PR TITLE
Adding Git hub access token

### PR DESCRIPTION
**Pull request by HazalCiplak**
https://github.com/elifesciences/data-hub-issues/issues/446

### CODE REVIEW

File: data_pipeline/elife_article_xml/elife_article_xml_pipeline.py [hunk 1]
```diff
@@ -29,8 +29,16 @@ class GitHubRateLimitError(requests.RequestException):
...
 
 
-def get_json_response_from_url(url: str) -> Any:
-    response = requests.get(url=url)
+def get_json_response_from_url(
+    url: str,
+    headers: Mapping[str, str] = None,
+) -> Any:
+    if headers:
+        LOGGER.info("API call with access token...")
```
Comment (de-code) [inline data_pipeline/elife_article_xml/elife_article_xml_pipeline.py]: Maybe nitpicking, but here we just know whether headers have been configured, not whether it using an access token. If you pass in the `printable_mapping`, then we could log it. (Also together with the URL would probably be useful)

File: sample_data_config/elife-article-xml/elife-article-xml.config.yaml [hunk 2]
```diff
@@ -3,6 +3,11 @@ elifeArticleXml:
     source:
       gitRepoUrl: 'https://api.github.com/repos/elifesciences/elife-article-xml/git/trees/master'
       directoryName: 'articles'
+      headers:
+        parametersFromFile:
+          # parameters will be skipped if referenced file does not exist
+          - parameterName: x-api-key
```
Comment (de-code) [inline sample_data_config/elife-article-xml/elife-article-xml.config.yaml]: I believe the parameter name is `Authorization`. And the value is not just the API token itself, but `token <TOKEN>`.
i.e. that would need to be reflected in the secret content, and its name.

Perhaps that is one disadvantage of the more generic "headers".

File: data_pipeline/elife_article_xml/elife_article_xml_pipeline.py [hunk 3]
```diff
@@ -29,8 +29,16 @@ class GitHubRateLimitError(requests.RequestException):
...
+        response = requests.get(url=url, headers=headers)
```
Comment (de-code) [inline data_pipeline/elife_article_xml/elife_article_xml_pipeline.py]: I believe you can always call it this way, even if `headers` is `None`.
(i.e. you don't need the condition)

File: sample_data_config/elife-article-xml/elife-article-xml.config.yaml (hunk 2 above)
Comment (HazalCiplak) [reply sample_data_config/elife-article-xml/elife-article-xml.config.yaml]: Yes I was looking for this, just find out and read you comment :D

**Review by de-code — APPROVED** [review]

**Pull request merged**